utils/azure_users.py

```diff
@@ -89,4 +74,42 @@ def to_azure_user(item) -> AzureUser:
 
         assert 200 == response.status_code
         assert 'value' in response.json()
-        return [to_azure_user(item) for item in response.json()['value']]
+        return [self.to_azure_user(item) for item in response.json()['value']]
+
+    def update_user_role(self, id, role):
+        headers = {
+            'Content-type': 'application/json',
+            'Accept': 'application/json',
+        }
+        endpoint = "{endpoint}/users/{user_id}?api-version=1.6".format(
+            endpoint=self.config.ENDPOINT, user_id=id
+        )
+        data = {'extension_1d76efa96f604499acc0c0ee116a1453_role': role}
+        response = requests.patch(
+            endpoint,
+            auth=BearerAuth(self.access_token),
+            data=json.dumps(data),
+            headers=headers,
+        )
+        assert 204 == response.status_code
+
+        response = requests.get(endpoint, auth=BearerAuth(self.access_token))
+        assert 200 == response.status_code
+
+        return self.to_azure_user(response.json())
+
+    def to_azure_user(self, item) -> AzureUser:
+        there_is_email = len(item['otherMails']) > 0
+        there_is_role = (
+            'extension_1d76efa96f604499acc0c0ee116a1453_role' in item
+        )
+
+        id = item['objectId']
+        name = item['displayName']
+        email = item['otherMails'][0] if there_is_email else ''
+        role = (
+            item['extension_1d76efa96f604499acc0c0ee116a1453_role']
```

Comment:
I see this piece of code thrice in this file. It would be great to extract it to a shared place. 
Yo can read the [DRY](https://github.com/HugoMatilla/The-Pragmatic-Programmer#7-the-evils-of-duplication) principle

Reply:
In addition I would extract  the 1d76efa96f604499acc0c0ee116a1453 hash as a env variable since it eventually might change.

Reply:
I moved into a variable of the class. May I argue against use it as a env variable? I consider that as a variable is enough.